camera_control/camera_control/camera/realsense.py

```diff
@@ -1,98 +1,132 @@
-        depth_scale = depth_sensor.get_depth_scale()
+        # Post-processing
+        self.align = (
+            rs.align(rs.stream.color) if self.align_depth_to_color_frame else None
+        )
+        self.temporal_filter = rs.temporal_filter()
+        # self.spatial_filter = rs.spatial_filter()  # Doesn't seem to help much. Disabling for now.
+        self.hole_filling_filter = rs.hole_filling_filter()
 
-        # We will be removing the background of objects more than
-        #  clipping_distance_in_meters meters away
-        clipping_distance_in_meters = 1  # 1 meter
-        self.clipping_distance = clipping_distance_in_meters / depth_scale
+    def set_exposure(self, exposure_ms):
+        color_sensor = self.profile.get_device().first_color_sensor()
+        if exposure_ms < 0:
+            color_sensor.set_option(rs.option.enable_auto_exposure, 1)
+        else:
+            # D435 has a minimum exposure time of 1us
+            exposure_us = max(1, round(exposure_ms * 1000))
+            color_sensor.set_option(rs.option.exposure, exposure_us)
 
     def get_frames(self):
         frames = self.pipeline.poll_for_frames()
-        if frames:
-            depth_frame = frames.get_depth_frame()
-            color_frame = frames.get_color_frame()
-            if not depth_frame or not color_frame:
-                return None
+        if not frames:
+            return None
 
-            return {"color": color_frame, "depth": depth_frame}
+        # Align depth frame to color frame if needed
+        if self.align:
+            frames = self.align.process(frames)
 
-    def get_pos_location(self, x, y, frame):
+        color_frame = frames.get_color_frame()
+        depth_frame = frames.get_depth_frame()
+        if not depth_frame or not color_frame:
+            return None
+
+        # Apply post-processing filters
+        depth_frame = self.temporal_filter.process(depth_frame)
+        depth_frame = self.hole_filling_filter.process(depth_frame)
+
+        # The various post processing functions return a generic frame, so we need
+        # to cast back to depth_frame
+        depth_frame = depth_frame.as_depth_frame()
+
+        return {"color": color_frame, "depth": depth_frame}
+
+    def get_pos_location(self, x, y, frames):
         """Given an x-y point in the color frame, return the x-y-z position with respect to the camera"""
-        depth_point = self._color_pixel_to_depth(x, y, frame)
+        depth_point = self._color_pixel_to_depth_pixel((x, y), frames["depth"])
         if not depth_point:
             return None
-        depth = frame["depth"].get_distance(
+        depth = frames["depth"].get_distance(
             round(depth_point[0]), round(depth_point[1])
         )
         pos_wrt_color = rs.rs2_deproject_pixel_to_point(
-            self.color_intrins, [x, y], depth
+            self.color_intrinsics, [x, y], depth
         )
         pos_wrt_color = np.array(pos_wrt_color) * 1000
```

Comment:
Why is there a *1000 here?

Reply:
I don't think it functionally affects anything as of now, but IMO we should stick to meters globally

Reply:
Thats fine by me, I'm a bit more partial to mm everywhere, but I don't really care.